### PR TITLE
Catch IllegalArgumentException if DataSaverStateChangeReceiver isn't …

### PR DESCRIPTION
**Pull request by mueller-ma**
…registered

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

**Comment by mueller-ma** [comment]
> doing the API level check

This is needed, because on unsupported api versions, the receiver is null.

> do it in onStop

Thanks! I didn't see that it registered in onStart, but unregistered in onPause.


I moved the registration logic to the receiver itself.

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt [hunk 1]
```diff
@@ -87,10 +87,8 @@ class WidgetListFragment : Fragment(), WidgetAdapter.ItemClickListener {
     private val suggestedCommandsFactory by lazy {
         SuggestedCommandsFactory(requireContext(), false)
     }
-    private val dataSaverChangeListener: BroadcastReceiver? = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+    private val dataSaverChangeListener by lazy {
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt]: Does the 'by lazy' get us anything here, given it'll be constructed unconditionally in onStart? I think the old variant actually was better, just the register call could've been made dependent on null status of the listener.

**Comment by mueller-ma** [comment]
Superseded by #1905

File: mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt (hunk 1 above)
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/ui/WidgetListFragment.kt]: I moved the unregistration to `onStop()` in #1905. The battery saver is available for all api levels the app supports, so no need to check for it in that PR.

**Pull request closed without merging**